File: Leetcode/155.cpp
class MinStack {
public:
    stack<int>element, minVal;
    MinStack() {
    }

    void push(int x) {
        element.push(x);
        if (minVal.empty() || x <= minVal.top()) {
            minVal.push(x);
        }
    }

    void pop() {
        if (element.empty()) return;
        if (element.top() == minVal.top()) {
            minVal.pop();
        }
        element.pop();
    }

    int top() {
        return element.top();
    }

    int getMin() {
        return minVal.top();
    }
};

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack* obj = new MinStack();
 * obj->push(x);
 * obj->pop();
 * int param_3 = obj->top();
 * int param_4 = obj->getMin();
 */

File: Leetcode/155.py
class MinStack:

    def __init__(self):
        """
        initialize your data structure here.
        """
        self.element, self.minVal = [], []


    def push(self, x: int) -> None:
        self.element.append(x)
        if not self.minVal or x <= self.minVal[-1]:
            self.minVal.append(x)

    def pop(self) -> None:
        if self.element is None:
            return
        if self.element[-1] == self.minVal[-1]:
            self.minVal.pop()
        self.element.pop()

    def top(self) -> int:
        return self.element[-1]

    def getMin(self) -> int:
        return self.minVal[-1]


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(x)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()
